nodebuilder/header/module_test.go

```diff
@@ -38,3 +42,41 @@ func TestConstructModule_StoreParams(t *testing.T) {
 	require.Equal(t, headerStore.Params.IndexCacheSize, cfg.Store.IndexCacheSize)
 	require.Equal(t, headerStore.Params.WriteBatchSize, cfg.Store.WriteBatchSize)
 }
+
+// TestConstructModule_ExchangeParams ensures that all passed via functional options
+// params are set in store correctly.
+func TestConstructModule_ExchangeParams(t *testing.T) {
+	cfg := DefaultConfig()
+	cfg.P2PExchange.WriteDeadline = time.Minute
+	cfg.P2PExchange.ReadDeadline = time.Minute * 2
+	cfg.P2PExchange.MinResponses = 10
+	cfg.P2PExchange.MaxRequestSize = 200
+	var exchange *p2p.Exchange
+	var exchangeServer *p2p.ExchangeServer
+
+	app := fxtest.New(t,
+		fx.Supply(modp2p.Private),
+		fx.Supply(modp2p.Bootstrappers{}),
+		fx.Provide(libp2p.New),
+		fx.Provide(func() datastore.Batching {
+			return datastore.NewMapDatastore()
+		}),
```

Comment:
nit: We might want to import the whole modp2p.ConstructModule` already

Reply:
wdym?

Reply:
ok. got your point. but `modp2p.ConstructModule ` requires additional params to be provided that brings complexity to the test.

Reply:
OKS